### ISSUE TITLE
[BUG] Re: 1.18 needs Java 17

### ISSUE THREAD

**Issue by fogdog-twilight** [AlexProgrammerDE/balena-minecraft-server#184]
It's a continuation of the issue below.
https://github.com/AlexProgrammerDE/balena-minecraft-server/issues/177

I have redeployed the latest server, but the console still says that java17 is required to start version 1.18.
Even if I set the ENABLE_UPDATE parameter to true, java17 was still not installed.
I can't find the openjdk directory in the /usr/local, and I don't know where java is installed.

My environment is as follows.
-Raspberry Pi 4 (using 64bit OS)
-Host OS version   : balenaOS 2.88.4
-Supervisor version: 12.11.21

![キャプチャ](https://user-images.githubusercontent.com/57118649/147043321-db06bb19-d325-4572-9d2c-53d7027d16db.PNG)

**Comment by UnderRTX** [comment]
The ENABLE_UPDATE parameter only updates minecraft version. Openjdk (java) has to be in version 17 from the beggining

**Comment by fogdog-twilight** [comment]
I don't think java17 is included in balenaos.
I can't run the latest minecraft.

**Comment by tlizri** [comment]
You have to download this project again (updated) and push it with balena-cli.

**Comment by AlexProgrammerDE** [comment]
Have you tried redeploying balena minecraft server?

**Comment by fogdog-twilight** [comment]
I redeproyed it with the latest version.

**Comment by taedium8vitae** [comment]
Just try to deploy it manually using the [Balena CLI tools](https://github.com/balena-io/balena-cli/blob/master/INSTALL.md) openjdk 17 is included in the latest release (debian bookworm).
Download the repository from github as a zip and use your terminal to push it to your fleet.
`cd YOURFOLDER`
then
`balena login`
and finally
`balena push YOURFLEETNAME`

**Comment by ComradeBv2** [comment]
How do you put openJDK 17 on the server?

**Comment by AlexProgrammerDE** [comment]
Mojang keeps requiring higher java versions with each release. I would recommend you to just redeploy the application as we ship java 17 with the images now.

**Comment by sumwhat1** [comment]
@AlexProgrammerDE I tried re-deploying it with one-click-deployment, and I even updated BalenaOS within balenaCloud but I'm still getting the error message. 

Thank for your continued dedication to this project 👍

**Comment by sumwhat1** [comment]
Ok, I figured out a fix.

If you're like me, you tried re-deploying and updating BalenaOS to no avail

![image](https://user-images.githubusercontent.com/86321696/150882837-3a56c4c0-6c05-4f27-9660-82257401fb46.png)

But it turns out that even with ENABLE_UPDATE = true, Paper.jar was not updating. So for some reason, the server was running 1.17 even though there was no such indication that was happening.

So you have to manually update Paper.jar by obtaining their latest JAR file, and then drag it into the serverfiles folder using SCP

To recap:
1. Re-deploy Project
2. Update BalenaOS using BalenaCloud-Actions
3. Replace Paper.Jar manually

After this, you should be good to go. Make sure to backup any important server files in case you need to start from scratch.

**Comment by ArnaldoScheibe** [comment]
> 

I had same problem and this fix to me.

Thank you for you post @sumwhat1

**Comment by sumwhat1** [comment]
> Hi, I have a couple of questions to ask.
> 
> Will I lose all data if I redeploy the project?

IIRC yes you will, best to backup all files in the server folder before redeploying the project.

> If I lose all the data how can I restore the data as before?

Just copy the server folder back into the same place you backed it up from earlier using SCP.

> Do I just need to back up all plug-ins, configuration files and worlds (to transfer them to the server later) or do I need to back up the entire `serverfiles` folder?

The entire folder is likely your best bet.

**Comment by rikvik2006** [comment]
Thanks for being so quick to respond 

> > Hi, I have a couple of questions to ask.
> > Will I lose all data if I redeploy the project?
> 
> IIRC yes you will, best to backup all files in the server folder before redeploying the project.
> 
> > If I lose all the data how can I restore the data as before?
> 
> Just copy the server folder back into the same place you backed it up from earlier using SCP.
> 
> > Do I just need to back up all plug-ins, configuration files and worlds (to transfer them to the server later) or do I need to back up the entire `serverfiles` folder?
> 
> The entire folder is likely your best bet.
